influxdb3_processing_engine/src/plugins.rs

```diff
@@ -142,12 +184,54 @@ mod python_plugin {
 
             Ok(())
         }
+
+        pub(crate) async fn run_schedule_plugin(
+            &self,
+            mut receiver: Receiver<PluginEvent>,
+            schedule: String,
+            time_provider: Arc<dyn TimeProvider>,
+        ) -> Result<(), Error> {
+            let schedule = Schedule::from_str(schedule.as_str())?;
+            let mut runner = ScheduleTriggerRunner::new(schedule, Arc::clone(&time_provider));
+
+            loop {
+                let Some(next_run_instant) = runner.next_run_time() else {
+                    break;
+                };
+
+                tokio::select! {
+                    _ = time_provider.sleep_until(next_run_instant) => {
+                        let Some(schema) = self.write_buffer.catalog().db_schema(self.db_name.as_str()) else {
+                            return Err(Error::MissingDb);
+                        };
+                        runner.run_at_time(self, schema).await?;
+                    }
+                    event = receiver.recv() => {
+                        match event {
+                            None => {
+                                warn!(?self.trigger_definition, "trigger plugin receiver closed");
+                                break;
+                            }
+                            Some(PluginEvent::WriteWalContents(_)) => {
+                                debug!("ignoring wal contents in cron plugin.")
```

Comment:
This is unexpected so maybe should be a `warn`? Can fix in a followup.

Reply:
Actually this will be happening. Right now we send the Wal contents to all plugins, without doing any up-front work to figure out whether they actually need the message. For WalRows, this means we don't check if there is data on a relevant table, and for scheduled plugins all of it is irrelevant.